modules/datadict/php/datadict.class.inc

```diff
@@ -58,8 +58,26 @@ class Datadict extends \DataFrameworkMenu
      */
     public function getFieldOptions() : array
     {
+        $instruments     = \Utility::getAllInstruments();
+        $dictInstruments = array();
+        foreach ($instruments as $instrument => $name) {
+            // Since the testname does not always match the table name in the
+            // database we need to instantiate the object to get the table name
+            $iObj = \NDB_BVL_Instrument::factory(
```

Comment:
This is going to cause the datadict to crash if you have an invalid instrument that throws an exception (ie. something invalid in test_names, maybe because you're switching between branches while developing an instrument or debugging an old/new version of LORIS..) . Can you put this in a try/catch block?

Reply:
what should I put in the catch ?
1. throw exception ? (defeats the purpose here I guess lol)
2. skip ?
3. default to testname ?

Reply:
Probably skip. I don't think defaulting to the test name is likely to result in any matches since the instrument is invalid, but either option 2 or 3. Maybe write to the error log too.

Reply:
@driusan I made the changes you requested. I have not tested however can you make sure to review carefully ?